src/main/java/htsjdk/samtools/filter/ReadNameFilter.java

```diff
@@ -104,19 +94,8 @@ public boolean filterOut(final SAMRecord record) {
      */
     @Override
     public boolean filterOut(final SAMRecord first, final SAMRecord second) {
-        if (includeReads) {
-            if (readNameFilterSet.contains(first.getReadName()) &&
-                readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        } else {
-            if (!readNameFilterSet.contains(first.getReadName()) &&
-                !readNameFilterSet.contains(second.getReadName())) {
-                return false;
-            }
-        }
-
-        return true;
+        return (readNameFilterSet.contains(first.getReadName()) &&
+                readNameFilterSet.contains(second.getReadName())) != includeReads;

```

Comment:
I agree that for the two-read version this is changing the behavior, and while I should have advertised it more clearly, I think that it's the right thing. that said, I'll keep it as it was.